src/lib/Cashlink.ts

```diff
@@ -201,12 +207,13 @@ export default class Cashlink {
                 if (balance) {
                     // Handle recharged/reused cashlink
                     if (!pendingClaimingTx) newState = CashlinkState.UNCLAIMED;
-                    break; // If a balance is detected on the cashlink, it cannot be CLAIMED.
+                    break; // If a balance is detected on the cashlink, it cannot be in CLAIMED state.
                 }
                 if (knownClaimingTx) {
                     newState = CashlinkState.CLAIMED;
                 }
             case CashlinkState.CLAIMED:
+                // Detect cashlink re-use
```

Comment:
I would specifically also mention rebranching because it took me a time to figure out whether the whole logic would function correctly on a rebranch.
Mentioning this here would take away that doubt.

Reply:
Does it work on rebranch? I did not design it for that. How is a rebranch special?

Reply:
I think it works.
On a rebranch we have to handle the case that our funding or claiming tx were included in a block that is not part of the main branch anymore after rebranching. The transactions are then added back to the mempool and pending again and thus a cashlink might be going back from any state to any previous state.
As all cases fall through to the `CLAIMED` case were a check for pending transactions happens that resets the state to an earlier state, I expect this to behave correctly on a rebranch.
There is a `break` in the `CLAIMING` if there is a balance, which is alright as if there is a balance after the rebranch, we won't need to go back to `CHARGING`.
The only thing that is not handled I think is going back to `UNCHARGED` in case the pending transactions expire. That's at current network load a pretty uncommon case, but if we want to, we might handle this too.

Reply:
There is a check when in CHARGING state:
```
                if (!balance && !pendingFundingTx) {
                    // Handle expired/replaced funding tx
                    newState = CashlinkState.UNCHARGED;
                    // Not break;ing here, because we need to see if the cashlink is already CLAIMED.
                }
```

I think this will reset it to UNCHARGED in the next check after the initial after-rebranch check.